interchaintest/misbehaviour_test.go

```diff
@@ -0,0 +1,320 @@
+	"github.com/strangelove-ventures/interchaintest/v7/ibc"
+	"github.com/strangelove-ventures/interchaintest/v7/testreporter"
+	"github.com/strangelove-ventures/interchaintest/v7/testutil"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+)
+
+func TestScenarioMisbehaviourDetection(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+
+	t.Parallel()
+
+	numVals := 1
+	cf := interchaintest.NewBuiltinChainFactory(zaptest.NewLogger(t), []*interchaintest.ChainSpec{
+		{Name: "gaia", Version: "v9.0.0-rc1", NumValidators: &numVals, ChainConfig: ibc.ChainConfig{ChainID: "chain-a", GasPrices: "0.0uatom"}},
+		{Name: "gaia", Version: "v9.0.0-rc1", NumValidators: &numVals, ChainConfig: ibc.ChainConfig{ChainID: "chain-b", GasPrices: "0.0uatom"}}},
+	)
+
+	chains, err := cf.Chains(t.Name())
+	require.NoError(t, err)
+	chainA, chainB := chains[0].(*cosmos.CosmosChain), chains[1].(*cosmos.CosmosChain)
+
+	ctx := context.Background()
+	client, network := interchaintest.DockerSetup(t)
+
+	rf := relayertest.NewRelayerFactory(relayertest.RelayerConfig{InitialBlockHistory: 50})
+	r := rf.Build(t, client, network)
+
+	const pathChainAChainB = "chainA-chainB"
+
+	ic := interchaintest.NewInterchain().
+		AddChain(chainA).
+		AddChain(chainB).
+		AddRelayer(r, "relayer").
+		AddLink(interchaintest.InterchainLink{
+			Chain1:           chainA,
+			Chain2:           chainB,
+			Relayer:          r,
+			Path:             pathChainAChainB,
+			CreateClientOpts: ibc.CreateClientOptions{TrustingPeriod: "15m"},
+		})
+
+	rep := testreporter.NewNopReporter()
+	eRep := rep.RelayerExecReporter(t)
+
+	require.NoError(t, ic.Build(ctx, eRep, interchaintest.InterchainBuildOptions{
+		TestName:         t.Name(),
+		Client:           client,
+		NetworkID:        network,
+		SkipPathCreation: false,
+	}))
+
+	t.Cleanup(func() {
+		_ = ic.Close()
+	})
+
+	// create a new user account and wait a few blocks for it to be created on chain
+	user := interchaintest.GetAndFundTestUsers(t, ctx, "user-1", 10_000_000, chainA)[0]
+	err = testutil.WaitForBlocks(ctx, 5, chainA)
+
+	// Start the relayer
+	require.NoError(t, r.StartRelayer(ctx, eRep, pathChainAChainB))
+
+	t.Cleanup(
+		func() {
+			err := r.StopRelayer(ctx, eRep)
+			if err != nil {
+				panic(fmt.Errorf("an error occured while stopping the relayer: %s", err))
+			}
+		},
+	)
+
+	// query latest height on chainB
+	latestHeight, err := chainB.Height(ctx)
+	require.NoError(t, err)
+
+	// query header at height on chainB
+	h := int64(latestHeight)
+	header, err := queryHeaderAtHeight(ctx, t, h, chainB)
+	require.NoError(t, err)
+
+	// query tm client state on chainA
+	const clientID = "07-tendermint-0"
+
+	cmd := []string{
+		chainA.Config().Bin, "q", "ibc", "client", "state", clientID,
+		"--node", chainA.GetRPCAddress(),
+		"--home", chainA.HomeDir(),
+		"--chain-id", chainA.Config().ChainID,
+		"--output", "json",
+	}
+
+	stdout, stderr, err := chainA.Exec(ctx, cmd, nil)
```

Comment:
This `cmd` could be simplified by using `chainA.Validators[0].ExecQuery`

Reply:
Oh nice, I thought we had something like this but was struggling to find it.